### ISSUE TITLE
improve the bonds query

### ISSUE THREAD

**Issue by tzemanovic** [anoma/namada#43]

**Comment by brentstone** [comment]
> @brentstone can you check on this / take this on as part of the PoS work?

Yeah, I noticed this recently and didn't realize there was an open issue for it. Sounds good.

**Comment by brentstone** [comment]
Perhaps use a lazy hash set or similar to store all addresses that have delegations to a validator (including self), then querying all delegations becomes much simpler.
